### PR TITLE
[WIP] Fix #2890: Make Tests Pass Again

### LINKED ISSUE #2890: Make tests pass again

### PR DESCRIPTION
**Pull request by jaredsilver**
**Note: this PR also fixes some bugs in the student profile activity component that I noticed while fixing the tests. We should test/verify that this works properly before merging/deploying this. (It might cause issues with the Lessons activities.)**

- [x] Remove legacy tests
- [x] Remove any legacy code you come across
- [x] Fix tests that are broken
- [x] Update snapshots
- [x] Fix Travis (_shakes fist at Travis_)

Separate PR:
- [ ] Fix linting errors 😱

## MERGE #2901 FIRST

### CODE REVIEW

**Comment by ddmck** [comment]
Can you restore the ELL announcement mini as we will need it the future for the resources section.

File: client/app/bundles/HelloWorld/components/dashboard/ell_diagnostic_announcement_mini.jsx [hunk 1]
```diff
@@ -1,30 +0,0 @@
-import React from 'react'
```
Comment (ddmck) [inline client/app/bundles/HelloWorld/components/dashboard/ell_diagnostic_announcement_mini.jsx]: Restore this file please.

Comment (jaredsilver) [reply client/app/bundles/HelloWorld/components/dashboard/ell_diagnostic_announcement_mini.jsx]: 👍

**Comment by jaredsilver** [comment]
Can we chat at some point about how we want our linter to be configured? Right now, our code is a mess relative to the rules we have in place, and we're actually running into an issue where Travis is erroring because we have more than 4mb worth of errors! 😱

**Pull request merged**